openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py

```diff
@@ -88,39 +97,75 @@ def tearDown(self):
+                                expected_tree,
+                                expected_pathnames,
+                                expected_conns,
+                                expected_abs2prom,
+                                expected_declare_partials,
+                                expected_driver_name,
+                                expected_design_vars_names,
+                                expected_responses_names
+                                ):
 
-        # check expected model tree
-        self.assertDictEqual(model_viewer_data['tree'], self.expected_tree)
+        self.assertDictEqual(model_viewer_data['tree'], expected_tree)
 
         # check expected system pathnames
         pathnames = model_viewer_data['sys_pathnames_list']
-        self.assertListEqual(sorted(pathnames), self.expected_pathnames)
+        self.assertListEqual(sorted(pathnames), expected_pathnames)
 
         # check expected connections, after mapping cycle_arrows indices back to pathnames
-        connections = sorted(model_viewer_data['connections_list'], key=lambda x: (x['tgt'], x['src']))
+        connections = sorted(model_viewer_data['connections_list'],
+                             key=lambda x: (x['tgt'], x['src']))
+        expected_conns = sorted(expected_conns,
+                             key=lambda x: (x['tgt'], x['src']))
+        self.assertEqual(len(connections), len(expected_conns))
         for conn in connections:
             if 'cycle_arrows' in conn and conn['cycle_arrows']:
                 cycle_arrows = []
                 for src, tgt in conn['cycle_arrows']:
                     cycle_arrows.append(' '.join([pathnames[src], pathnames[tgt]]))
                 conn['cycle_arrows'] = sorted(cycle_arrows)
-        self.assertEqual(len(connections), len(self.expected_conns))
-        for c, ex in zip(connections, self.expected_conns):
+        for c, ex in zip(connections, expected_conns):
             self.assertEqual(c['src'], ex['src'])
             self.assertEqual(c['tgt'], ex['tgt'])
             self.assertEqual(c.get('cycle_arrows', []), ex.get('cycle_arrows', []))
 
         # check expected abs2prom map
-        self.assertDictEqual(model_viewer_data['abs2prom'], self.expected_abs2prom)
+        self.assertListEqual(sorted(model_viewer_data['abs2prom']), sorted(expected_abs2prom))
+
+        # check expected declare_partials_list
+        self.assertListEqual(sorted(model_viewer_data['declare_partials_list']),
+                             sorted(expected_declare_partials))
+
+        self.assertEqual(model_viewer_data['driver']['name'], expected_driver_name)
+
+        self.assertListEqual( sorted(dv for dv in model_viewer_data['design_vars']), sorted(expected_design_vars_names))
+        self.assertListEqual( sorted(resp for resp in model_viewer_data['responses']), sorted(expected_responses_names))
+
+    def test_model_viewer_has_correct_data_from_problem(self):
+        """
```

Comment:
I remember that there is a weird interaction with testflo (or maybe unittest) and 3-quote docstrings, where the title gets replaced by the docstring in the output.  That's why we (or I) usually use "#" single line comments.

Reply:
Thanks. I changed those. But there are probably 100 more of those in our tests. Should we make a story to change them?

Reply:
Oh, if there are that many, maybe I am wrong and they (bret or unittest devs) have fixed this.

Reply:
I do remember an issue like that also but it must be fixed.